app/src/main/java/com/jerboa/feed/FeedController.kt

```diff
@@ -0,0 +1,76 @@
+package com.jerboa.feed
+
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.snapshots.SnapshotStateList
+
+open class FeedController<T> {
```

Comment:
I was thinking that too but all our "update" methods must use this safe update method. I do not want to be able accidentally use the builtin update that will throw exceptions for items that it doesn't find. All the exposed methods currently are safe, none will throw exceptions. And I do not allow access to SnapShotState directly where one accidentally could do `snap[i] =`

The few extra copies are worth it imo.

Reply:
Would it be easier to just use inheritance with an override, [described here](https://kotlinlang.org/docs/inheritance.html).

IE something like 

```kotlin
open class JerboaSnapshotStateList(): SnapshotStateList {
  final override fun update(...) {
```

We can't be the only ones dealing with this problem also, I'd be surprised if there's not already a library that handles this case.

If those aren't possible, then its no problem, just seems like a lot of unecessary code.

Reply:
>  just seems like a lot of unecessary code.

Its four lines
```
    fun add(item: T) = items.add(item)
    fun remove(item: T) = items.remove(item)
    fun clear() = items.clear()
    fun addAll(newItems: List<T>) = items.addAll(newItems)
```

I didn't do inheritance because then I would expose a lot different methods that I don't want either. Plus then they would still be able to `snap[]` in a unsafe matter.

Reply:
mmmk, that's no problem then.